x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/BinarySpatialFunction.java

```diff
@@ -80,6 +81,29 @@ public void writeTo(StreamOutput out) throws IOException {
         // The CRS type is re-resolved from the combination of left and right fields, and also not necessary to serialize
     }
 
+    /**
+     * Mark the function as expecting the specified fields to arrive as doc-values.
+     */
+    public abstract BinarySpatialFunction withDocValues(boolean foundLeft, boolean foundRight);
+
+    @Override
+    public int hashCode() {
+        // NB: the hashcode is currently used for key generation so
+        // to avoid clashes between aggs with the same arguments, add the class name as variation
+        return Objects.hash(getClass(), children(), leftDocValues, rightDocValues);
```

Comment:
This is covered by the getClass method, since pointOnly is true for one class and false for all others. While we could express pointsOnly differently (eg. a method on the class, so more obviously tied to the class itself), I would rather do that in a different PR since it is unrelated to the purpose of this PR (ie. a pre-existing concern). Since this was a trivial refactoring that did not change the behaviour of equals/hashcode, just moved it up to the parent class, lets consider that later. My preference would be when we implement distance functions for shapes, which will require a change to this logic anyway.